src/main/java/jewellery/inventory/service/SaleService.java

```diff
@@ -141,19 +163,23 @@ private void updateProductOwnersAndSale(
   }
 
   private List<ProductPriceDiscount> getProductsFromSaleRequestDto(SaleRequestDto saleRequestDto) {
-    return saleRequestDto.getProducts().stream()
-        .map(
-            productDto -> {
-              ProductPriceDiscount productPriceDiscount = new ProductPriceDiscount();
-              productPriceDiscount.setProduct(productService.getProduct(productDto.getProductId()));
-              productPriceDiscount.setDiscount(productDto.getDiscount());
-              return productPriceDiscount;
-            })
-        .toList();
+    if (saleRequestDto.getProducts() != null) {
+      return saleRequestDto.getProducts().stream()
+          .map(
+              productDto -> {
+                ProductPriceDiscount productPriceDiscount = new ProductPriceDiscount();
+                productPriceDiscount.setProduct(
+                    productService.getProduct(productDto.getProductId()));
+                productPriceDiscount.setDiscount(productDto.getDiscount());
+                return productPriceDiscount;
+              })
+          .toList();
+    }
+    return new ArrayList<>();
   }
 
-  private void deleteSaleIfProductsIsEmpty(Sale sale) {
-    if (sale.getProducts().isEmpty()) {
+  private void deleteSaleIfProductsAndResourcesAreEmpty(Sale sale) {
+    if (sale.getProducts().isEmpty() && sale.getResources().isEmpty()) {
       logger.info("Deleting sale with ID: {} since the products list is empty.", sale.getId());
```

Comment:
We should also change the logger message to point out that both  resources and products are empty

Reply:
done!